BQ25895M.device.nut

```diff
@@ -0,0 +1,302 @@
+        
+    } 
+     
+    // Clear the enable charging bit, device will not charge until enableCharging() is called again
+    function disableCharging(){
+        
+        local rd = _getReg(BQ25895M_REG03);
+        rd = rd & ~(1<<4); // clear CHG_CONFIG bits
+        
+        _setReg(BQ25895M_REG03, rd);
+        
+    } 
+    
+    // Set target battery voltage
+    function setChargeVoltage(VREG){ 
+        
+        // Check that input is within accepted range
+        if (VREG < 3.840) VREG = 3.840;
+        else if (VREG > 4.608) VREG = 4.608;
+        
+        local rd = _getReg(BQ25895M_REG06);
+        rd = rd & ~(0xFC); // clear current limit bits
+        rd = rd | (0xFC & ((((VREG*1000) - 3840)/16).tointeger()) << 2);
+        
+        _setReg(BQ25895M_REG06, rd);
+        
+    }
+
+    // Set fast charge current
+    function setChargeCurrent(ICHG){
+        
+        // Check that input is within accepted range
+        if (ICHG < 0) ICHG = 0;
+        else if (ICHG > 5056) ICHG = 5056;
+        
+        local rd = _getReg(BQ25895M_REG04);
+        rd = rd & ~(0x7F); // clear current limit bits
+        rd = rd | (0x7F & ICHG/64); // set ILIM
+    
+        _setReg(BQ25895M_REG04, rd);
+        
+    }
+    
+    // Returns the target battery voltage
+    function getChargeVoltage(){
+        
+        local rd = _getReg(BQ25895M_REG06);
+        local ChrgVlim = ((rd>>2)*16)+3840;
+
+        return ChrgVlim;
+        
+    }
+    
+    // Returns the battery voltage based on the ADC conversion
+    function getBatteryVoltage(){
+        
+        _convStart(); // Kick ADC
+        
+        local rd = _getReg(BQ25895M_REG0E);
+        local BATV =(2304+(20*(rd&0x7f)));
+        
+        return BATV;
+        
+    }
+     
+    // Returns the VBUS voltage based on the ADC conversion, this is the input voltage
+    function getVBUSVoltage(){ 
+        
+        _convStart(); // Kick ADC
+
+        local rd = _getReg(BQ25895M_REG11);
+        local VBUSV = (2600+(100*(rd&0x7f)))
+        
+        return VBUSV; 
+    }
+    
+    // Returns the system voltage based on the ADC conversion
+    function getSystemVoltage(){
+    
+        _convStart(); // Kick ADC
+    
+        local rd = _getReg(BQ25895M_REG0F);
+        local SYSV = (2304+(20*(rd&0x7f)));
+        
+        return SYSV;
+        
+    }
+
+    // Returns the measured charge current based on the ADC conversion
+    function getChargingCurrent(){
+        
+        _convStart(); // Kick ADC
+        
+        local rd = _getReg(BQ25895M_REG12);
+        local ICHGR = (50*(rd&0x7f))
+        
+        return ICHGR;  
+    }
+    
+    // Returns the charging status: Not Charging, Pre-charge, Fast Charging, Charge Termination Good
+    function getChargingStatus(){
+        
+        local chargingStatus;
+        
+        local rd = _getReg(BQ25895M_REG0B)
+        rd = rd & 0x18; 
+        
+        local mode = "";
+        
+        switch(rd) {
```

Comment:
If you are using an enum for charging status you should use the enum here. In fact I would update the enum to 

enum BQ25895M_CHARGING_STATUS {
    NOT_CHARGING                         = 0x00,
    PRE_CHARGE                              = 0x08,
    FAST_CHARGING                        = 0x10,
    CHARGE_TERMINATION_DONE = 0x18
}

Then you can eliminate the switch statement entirely. The example usage would look something like the following:  

local status = charger. getChargingStatus();
switch(status) {
     case BQ25895M_CHARGING_STATUS.NOT_CHARGING : 
          // Do something
          break;
     case BQ25895M_CHARGING_STATUS.PRE_CHARGE : 
          // Do something
          break;
     case BQ25895M_CHARGING_STATUS.FAST_CHARGING : 
          // Do something
          break;
     case BQ25895M_CHARGING_STATUS.CHARGE_TERMINATION_DONE : 
          // Do something
          break;
}